### DISCUSSION TITLE
Using @Retry on rest client interface?

### DISCUSSION THREAD

**Discussion by clotton** [quarkusio/quarkus#27445] [Q&A]
Hello,

Will this work?

@RegisterRestClient(configKey = "my-api")
@Retry(delay = 1, delayUnit = ChronoUnit.SECONDS, maxRetries = 5)
@ExponentialBackoff
public interface MyRestClient {
    ... my methods for get/post
}

Such that any rest api in MyRestClient will automatically be retried if an error is returned?

[a comment by an automated account is not shown]

**Comment by michalszynkiewicz** [comment]
Yes, IIRC, this will work.

**Comment by GeezerPelletier** [comment]
@michalszynkiewicz will this also work if the retries are configured on the methods? E.g.

@RegisterRestClient(configKey = "my-api")
public interface MyRestClient {

  @POST
  @Retry(delay = 1, delayUnit = ChronoUnit.SECONDS, maxRetries = 5)
  @ExponentialBackoff
  Response postMethod1(...);

  @POST
  @Retry(delay = 5, delayUnit = ChronoUnit.SECONDS, maxRetries = 99)
  @ExponentialBackoff
  Response postMethod2(...);
}

**Reply by qmuldiytsxi** [reply]
I had the exact same questions like above. Yes it works, both on annotated classes and annotated methods of rest clients. The only thing is that Quarkus doesn't print any logs of the retries e.g. in quarkus:dev with the default log level which is not so nice from my point of view. If you change the log level to TRACE you can see the retrying outgoing requests.
